### DISCUSSION TITLE
Is it possible to customize the default header transition for screens?

### DISCUSSION THREAD

**Discussion by andrew-levy** [software-mansion/react-native-screens#1865] [Ideas]
When used in a library like React Navigation, the default iOS header uses this default animation:

https://github.com/software-mansion/react-native-screens/assets/29075740/737d3616-f34b-4dd0-8228-f1d182110583

There are two problems with this: 
1. Not customizable (to my knowledge)
2. When a custom header is used (ie your own component), you aren't able to animate the header elements between screen transitions.

My goal is to create a "custom" transition animation based on the transition progress My questions are:

1. Is it possible to customize the header transition using this library? For example, could we pass in our own NavigationController that overrides some of these animations?
3. Is it possible to access the transition progress (in JS land) so that I can just animate certain header elements based on that progress value? i found this [implementation](https://gist.github.com/policante/33a2d11b0501052d230c1f213dcacde3) that sort of does this. Also, reanimated has some form of progress animations with shared element transitions https://twitter.com/piaskowyk/status/1683846036878393349?s=20

**Comment by andrew-levy** [comment]
please let me know if this isn't the correct place for this question.

**Comment by kkafar** [comment]
No, we currently do not expose any API for customising header transitions, we rely on purely native default behaviour here (I believe `stackAnimation` does not modify header transition here? I'm not 100% sure tbh :D). 

Also we do not notify JS side on transition progress. There is `activityState` however it is updated on transition start & end, thus not really suitable for driving animations. 

Such feature was not on my radar before, however as you brought it to my attention I'll write that down to my backlog. It is not a priority though.
